### PR TITLE
uapi: documentation for the uapi crate

### PR DESCRIPTION
**Pull request by pthierry-ledger**
Add clean documentation for UAPI crate

- [x] add ::uapi module interface documentation
- [x] add module level documentation for all modules
- [x] add ::syscall module interface documentation
- [x] add ::svc_exchange interface documentation
- [x] add ::systypes module itnerface documentation

### CODE REVIEW

File: uapi/uapi/uapi.rs [hunk 1]
```diff
@@ -57,6 +142,41 @@ pub fn get_cycles() -> Result<u64, Status> {
...
+pub fn get_process_handle(process: ProcessLabel, _handle: *mut taskh_t) -> Result<taskh_t,Status> {
+    match syscall::sys_get_process_handle(process) {
+        Status::Ok => (),
+        any_err => return Err(any_err),
+    };
+    // FIXME: we might use copy_to_user here
+    let exch_area = unsafe { &mut SVC_EXCHANGE_AREA[..4] };
+    let result = u32::from_ne_bytes(exch_area.try_into().map_err(|_| Status::Invalid)?);
+    Ok(result)
+}
```
Comment (fvalette-ledger) [inline uapi/uapi/uapi.rs:178]: `_handle` is not used

Comment (pthierry-ledger) [reply uapi/uapi/uapi.rs:182]: Right. Will be removed that, as, as described in the UAPI page of the wiki, this part will be pushed up to a dedicated crate.

File: uapi/uapi/svc_exchange.rs [hunk 2]
```diff
@@ -3,7 +3,9 @@
 
 use crate::systypes::{EraseMode, EraseType, Status};
 
+/// SVC Exchange area len. TODO: to be kconfig-generated
 pub const SVC_EXCH_AREA_LEN: usize = 128; // TODO: replace by CONFIG-defined value
```
Comment (fvalette-ledger) [inline uapi/uapi/svc_exchange.rs:7]: Is this still used ?

**Review by fvalette-ledger — APPROVED** [review]

File: uapi/uapi/svc_exchange.rs (hunk 2 above)
Comment (pthierry-ledger) [reply uapi/uapi/svc_exchange.rs:7]: By now yes, but by nobody.... (extern C do not use this, shield-rs doesn't exist. This requires to be fixed though, but in another PR. Checking for issue

**Pull request merged**